src/coreComponents/constitutive/diffusion/ConstantDiffusion.hpp

```diff
@@ -0,0 +1,112 @@
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ConstantDiffusion.hpp
+ */
+
+#ifndef GEOS_CONSTITUTIVE_DIFFUSION_CONSTANTDIFFUSION_HPP_
+#define GEOS_CONSTITUTIVE_DIFFUSION_CONSTANTDIFFUSION_HPP_
+
+#include "constitutive/diffusion/DiffusionBase.hpp"
+
+namespace geos
+{
+namespace constitutive
+{
+
+/**
+ * @brief The update class for constant diffusion (does not do anything)
+ */
+class ConstantDiffusionUpdate : public DiffusionBaseUpdate
+{
+public:
+
+  /**
+   * @brief Constructor for the class performing the diffusion updates
+   * @param diffusivity the array of cell-wise diffusivities in the subregion
+   * @param dDiffusivity_dTemperature the array of cell-wise derivatives of diffusivities wrt temperature in the subregion
+   */
+  ConstantDiffusionUpdate( arrayView3d< real64 > const & diffusivity,
+                           arrayView3d< real64 > const & dDiffusivity_dTemperature )
+    : DiffusionBaseUpdate( diffusivity, dDiffusivity_dTemperature )
+  {}
+
+  GEOS_HOST_DEVICE
+  virtual void update( localIndex const k,
+                       localIndex const q,
+                       real64 const & temperature ) const override
+  { GEOS_UNUSED_VAR( k, q, temperature ); }
+
+};
+
+/**
+ * @brief The class for constant diffusion
+ */
+class ConstantDiffusion : public DiffusionBase
+{
+public:
+
+  /**
+   * @brief Constructor for the class storing constant diffusion
+   * @param[in] name the name of the class
+   * @param[in] parent pointer to the parent Group
+   */
+  ConstantDiffusion( string const & name, Group * const parent );
+
+  std::unique_ptr< ConstitutiveBase > deliverClone( string const & name,
+                                                    Group * const parent ) const override;
+
+  virtual void allocateConstitutiveData( dataRepository::Group & parent,
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  static string catalogName() { return "ConstantDiffusion"; }
+
+  virtual string getCatalogName() const override { return catalogName(); }
+
+  /// Type of kernel wrapper for in-kernel update
+  using KernelWrapper = ConstantDiffusionUpdate;
+
+  /**
+   * @brief Create an update kernel wrapper.
+   * @return the wrapper
+   */
+  KernelWrapper createKernelWrapper() const
+  {
+    return KernelWrapper( m_diffusivity,
+                          m_dDiffusivity_dTemperature );
+  }
+
+  struct viewKeyStruct : public DiffusionBase::viewKeyStruct
+  {
+    static constexpr char const * diffusivityComponentsString() { return "diffusivityComponents"; }
+  } viewKeys;
+
+protected:
+
+  virtual void postProcessInput() override;
+
+private:
+
+  /// default diffusivity in the subRegion
+  R1Tensor m_diffusivityComponents;
```

Comment:
Idk if I remember correctly but wasn't there a policy not to use `R1Tensor` anymore ?

Reply:
We only stopped using `R1Tensor` for intermediate variables in kernels due to some concerns about how well they optimize IIRC. We do use `R1Tensor` for reading from input for things that are "vector"-like (e.g. geometric points, permeability components, gravity, etc). This has the benefit of better conveying the intent to the user (via rST/schema), as well automatic validation of input size. That said, if the input of full-tensor quantities is addressed, this may change and we might just make them all `array1d` to allow inputting a variable number of values (1, 3 or 6). So I'm not necessarily suggesting reverting the change, just bringing it up for context.

Reply:
Thanks, I wasn't sure at all I recall it correctly. My bad sorry @francoishamon